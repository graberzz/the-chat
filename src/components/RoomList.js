import React from 'react';
import PropTypes from 'prop-types';

import RoomItem from './RoomItem/';

const RoomList = ({ rooms }) => (
  <ul>
      { rooms.map(room => 
        <RoomItem key={room.id} room={room} />)}
  </ul>
);

RoomList.propTypes = {
  rooms: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
  })),
};

export default RoomList;