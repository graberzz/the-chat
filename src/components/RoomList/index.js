import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import RoomItem from '../RoomItem';

const NoRooms = styled.div`
  text-align: center;
  color: #aaa;
  font-size: 2em;
`;

const RoomList = ({ rooms }) => (
  <ul>
      { rooms.length ? rooms.map(room => 
        <RoomItem key={room.id} room={room} />) :
        <NoRooms>no rooms yet...</NoRooms> }
  </ul>
);

RoomList.propTypes = {
  rooms: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
  })).isRequired,
};

export default RoomList;