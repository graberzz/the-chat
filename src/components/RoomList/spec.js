import React from 'react';
import { shallow } from 'enzyme';
import RoomList from './';

const testRooms = [
  {
    id: '1',
    name: 'roomone',
    creator: 'admin',
  },
  {
    id: '2',
    name: 'roomtwo',
    creator: 'admin',
  },
];

describe('<RoomList />', () => {
  it('matches snapshot', () => {
    expect(shallow(<RoomList rooms={testRooms} />)).toMatchSnapshot();
  });
});