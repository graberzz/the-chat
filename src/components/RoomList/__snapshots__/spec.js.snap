// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoomList /> matches snapshot 1`] = `
<ul>
  <RoomItem
    key="1"
    room={
      Object {
        "creator": "admin",
        "id": "1",
        "name": "roomone",
      }
    }
  />
  <RoomItem
    key="2"
    room={
      Object {
        "creator": "admin",
        "id": "2",
        "name": "roomtwo",
      }
    }
  />
</ul>
`;
